File: app/src/main/java/dev/radley/omgstarwars/network/model/Planet.java
package dev.radley.omgstarwars.network.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * Created by Oleur on 22/12/2014.
 * Planet model represents a large mass, planet or planetoid in the Star Wars Universe, at the time of 0 ABY.
 */
public class Planet extends SWModel implements Serializable {

    public String diameter;
    public String gravity;
    public String population;
    public String climate;
    public String terrain;

    @SerializedName("rotation_period")
    public String rotationPeriod;

    @SerializedName("orbital_period")
    public String orbitalPeriod;

    @SerializedName("surface_water")
    public String surfaceWater;

    @SerializedName("residents")
    public ArrayList<String> residentsUrls;

    @SerializedName("films")
    public ArrayList<String> filmsUrls;


    @Override
    public ArrayList<String> getPeople() {
        return residentsUrls;
    }

    @Override
    public ArrayList<String> getFilms() {
        return filmsUrls;
    }

    @Override
    public String getRelatedPeopleTitle() {
        return "Residents";
    }
}

File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/Planet.kt
package dev.radley.omgstarwars.models

import com.google.gson.annotations.SerializedName
import dev.radley.omgstarwars.utilities.FormatUtils

import java.io.Serializable
import java.util.ArrayList

/**
 * Planet model represents a large mass, planet or planetoid in the Star Wars Universe, at the time of 0 ABY.
 */
class Planet : SWModel(), Serializable {

    var diameter: String = ""
    var gravity: String = ""
    var population: String = ""
    var climate: String = ""
    var terrain: String = ""

    override val subtitle: String
        get() = "Population: " + FormatUtils.getFormattedNumber(population)

    @SerializedName("rotation_period")
    var rotationPeriod: String = ""

    @SerializedName("orbital_period")
    var orbitalPeriod: String = ""

    @SerializedName("surface_water")
    var surfaceWater: String = ""

    @SerializedName("residents")
    override var relatedPeople: ArrayList<String>? = null

    @SerializedName("films")
    override var relatedFilms: ArrayList<String>? = null

    override val relatedPeopleTitle: String
        get() = "Residents"
}
